src/main/steps/respondent/consent-to-application/ConsentMapper.ts

```diff
@@ -1,59 +1,42 @@
 import { CaseWithId } from '../../../app/case/case';
 import { Consent, Respondent, YesOrNo } from '../../../app/case/definition';
-import { fromApiDate } from '../../../app/case/from-api-format';
-import { toApiDate } from '../../../app/case/to-api-format';
-import type { AppRequest } from '../../../app/controller/AppRequest';
-
-export const setConsentDetails = (respondent: Respondent, req: AppRequest): Respondent => {
-  let consentFromResponsent: Consent;
-  let reasonForNotConsenting = '';
-  let courtOrderDetails = '';
-  if (req.session.userCase.doYouConsent === YesOrNo.NO) {
-    reasonForNotConsenting = req.session.userCase.reasonForNotConsenting!;
-  }
-  if (req.session.userCase.courtPermission === YesOrNo.YES) {
-    courtOrderDetails = req.session.userCase.courtOrderDetails!;
+
+export const prepareRequest = (req: CaseWithId): Consent => {
+  const { doYouConsent, courtPermission, reasonForNotConsenting, courtOrderDetails } = req;
+
+  const request: Consent = {};
+
+  Object.assign(request, {
+    consentToTheApplication: doYouConsent,
+    permissionFromCourt: courtPermission,
+    noConsentReason: reasonForNotConsenting,
+    courtOrderDetails,
+  });
+
+  // data cleanup
+  if (doYouConsent === YesOrNo.YES) {
+    delete request.noConsentReason;
   }
-  if (respondent?.value?.response && respondent?.value?.response?.consent) {
```

Comment:
Why is the existing data cleanup conditions removed? Where have you handled all these in the prepareRequest?

Reply:
We are not dealing with the response.value.response in the prepareRequest method right? The only single responsibility of prepareRequest is to take the values from request, clean it up if needed and then return the request.